Velaptor/InternalExtensionMethods.cs

```diff
@@ -1479,4 +1477,109 @@ public static int IndexOf<T>(this IEnumerable<T> items, Predicate<T> predicate)
 
         return -1;
     }
+
+    /// <summary>
+    /// Returns the index of the first occurence of an item that matches the given <paramref name="predicate"/> result.
+    /// </summary>
+    /// <param name="items">The items to check.</param>
+    /// <param name="predicate">Indicates if the item index should be returned.</param>
+    /// <typeparam name="T">The type of item in list.</typeparam>
+    /// <returns>The index of the item that the <paramref name="predicate"/> returned <c>true</c> on.</returns>
+    /// <remarks>
+    ///     The iterating over the items will stop once the <paramref name="predicate"/> returns true.
+    /// </remarks>
+    public static int FirstItemIndex<T>(this Memory<T> items, Predicate<T> predicate)
+    {
+        var index = -1;
+
+        foreach (var item in items.Span)
+        {
+            index++;
+
+            if (predicate(item))
+            {
+                return index;
+            }
+        }
+
+        return -1;
+    }
+
+    /// <summary>
+    /// Returns the index of the first item that matches the given <paramref name="layer"/>.
+    /// </summary>
+    /// <param name="items">The items to check.</param>
+    /// <param name="layer">The layer to check for.</param>
+    /// <typeparam name="T">The type of <see cref="RenderItem{T}"/>.</typeparam>
+    /// <returns>
+    ///     The index of the item in the list of <paramref name="items"/>.
+    ///     <br/>
+    ///     The value of -1 will be returned if the layer can not be found.
+    /// </returns>
+    public static int FirstLayerIndex<T>(this Memory<RenderItem<T>> items, int layer)
+    {
+        for (var i = 0; i < items.Span.Length; i++)
+        {
+            if (items.Span[i].Layer == layer)
+            {
+                return i;
+            }
+        }
+
+        return -1;
+    }
+
+    /// <summary>
+    /// Gets the total number of items that are on the given <paramref name="layer"/>.
+    /// </summary>
+    /// <param name="items">The items to check.</param>
+    /// <param name="layer">The layer to check for.</param>
```

Comment:
Refactor to omit the "for" at the end. 
```suggestion
    /// <param name="layer">The layer to check.</param>
```